tests/run.go

```diff
@@ -620,9 +620,7 @@ func (t *test) run() {
 		os.Setenv("GOOS", runtime.GOOS)
 	}
 	if os.Getenv("GOARCH") == "" {
-		//os.Setenv("GOARCH", runtime.GOARCH)
-		// GOPHERJS.
-		os.Setenv("GOARCH", "js") // We're running this script natively, but the tests are executed with js architecture.
+		os.Setenv("GOARCH", runtime.GOARCH)
```

Comment:
Yes, tests/run.go is a copy of https://github.com/golang/go/blob/go1.8/test/run.go, with modifications to support GopherJS. Sorry I didn't make that more clear.